client/my-sites/email/email-providers-stacked-comparison/index.tsx

```diff
@@ -109,16 +133,7 @@ const EmailProvidersStackedComparison = ( {
 			} )
 		);
 
-		page(
-			emailManagementPurchaseNewEmailAccount(
-				selectedSite.slug,
-				selectedDomainName,
-				currentRoute,
-				null,
-				selectedEmailProviderSlug,
-				newIntervalLength
-			)
-		);
+		onIntervalLengthPathHandlerResolver( newIntervalLength );
```

Comment:
What do you think of passing this handler to this component? In that case we could give this handler all the props it needs like this:

```suggestion
		onIntervalLengthChange( {
			selectedSiteSlug,
			selectedDomainName,
			currentRoute,
			selectedEmailProviderSlug,
			newIntervalLength
		} );
```

And defines the handler at the top level like this:

```javascript
const changeIntervalLength = ( props ) => {
  page(
      domainAddEmailUpsell(
        props.selectedSiteSlug,
        props.selectedDomainName,
        props.newIntervalLength
      )
    );
};

<EmailProvidersStackedComparison
  comparisonContext="domain-upsell"
  isDomainInCart={ true }
  onIntervalLengthChange={ changeIntervalLength }
  selectedDomainName={ domain }
  selectedIntervalLength={ selectedIntervalLength }
  source="domain-upsell"
>
```